docs/source/data/fetching-rest.mdx

```diff
@@ -1,77 +1,114 @@
-| [`SQLDataSource`](https://github.com/cvburgess/SQLDataSource)  | Community | SQL databases (via [Knex.js](http://knexjs.org/)) |
-| [`MongoDataSource`](https://github.com/GraphQLGuide/apollo-datasource-mongodb/) | Community | MongoDB |
-| [`CosmosDataSource`](https://github.com/andrejpk/apollo-datasource-cosmosdb) | Community | Azure Cosmos DB |
-| [`FirestoreDataSource`](https://github.com/swantzter/apollo-datasource-firestore) | Community | Cloud Firestore |
+```bash
+npm install @apollo/datasource-rest
+```
 
-If none of these implementations applies to your use case, you can create your own custom `DataSource` subclass.
+Your server should define a separate subclass of `RESTDataSource` for each REST API it communicates with. Here's an example of a `RESTDataSource` subclass that defines two data-fetching methods, `getMovie` and `getMostViewedMovies`:
+<MultiCodeBlock>
 
-> Apollo does not provide official support for community-maintained libraries. We cannot guarantee that community-maintained libraries adhere to best practices, or that they will continue to be maintained.
+```ts title="movies-api.ts"
+import { RESTDataSource } from '@apollo/datasource-rest';
 
-## Adding data sources to Apollo Server
+class MoviesAPI extends RESTDataSource {
+  baseURL = 'https://movies-api.example.com/';
+  constructor() {
+    super();
+  }
 
-You provide your `DataSource` subclasses to the `ApolloServer` constructor, like so:
+  async getMovie(id):Promise<Movie> {
+    return this.get(`movies/${encodeURIComponent(id)}`);
+  }
 
-```ts {4-9} title="index.js"
-const server = new ApolloServer({
+  async getMostViewedMovies(limit = '10'):Promise<Movie[]> {
+    const data = await this.get('movies', {
+      params: {
+        per_page: limit,
+        order_by: 'most_viewed',
+      },
+    });
+    return data.results;
+  }
+}
+```
+
+</MultiCodeBlock>
+
+You can extend the `RESTDataSource` class to implement whatever data-fetching methods your resolvers need. These methods can use built-in convenience methods (like `get` and `post`) to perform HTTP requests, helping you add query parameters, parse JSON results, and handle errors.
+
+## Adding subclasses to Apollo Server's context
+
+You can then provide your `RESTDataSource` subclasses to the `context` initialization function, like so:
+
+<MultiCodeBlock>
+
+```ts title="index.ts"
+//highlight-start
+interface ContextValue {
+  dataSources: {
+    moviesAPI: MoviesAPI;
+    personalizationAPI: PersonalizationAPI;
+  };
+}
+//highlight-end
+
+const server = new ApolloServer<ContextValue>({
   typeDefs,
   resolvers,
-  csrfPrevention: true,
-  cache: 'bounded',
-  plugins: [ApolloServerPluginLandingPageLocalDefault({ embed: true })],
-  dataSources: () => {
+});
+
+const { url } = await startStandaloneServer(server, {
+  context: async ({ req }) => {
     return {
```

Comment:
We want to pass the Apollo Server cache to the data sources when we instantiate them. Assuming we haven't overridden the constructor in our subclass implementations, they'll share the same signature as `RESTDataSource` itself.

```ts
const { url } = await startStandaloneServer(server, {
  context: async ({ req }) => {
    const { cache } = server;
    return {
      //highlight-start
      // We create new instances of our data sources with each request
      dataSources: {
        moviesAPI: new MoviesAPI({ cache }),
        personalizationAPI: new PersonalizationAPI({ cache }),
      },
      //highlight-end
    };
  },
});
```